server/log_rpc_server.go

```diff
@@ -549,10 +553,18 @@ func (t *TrillianLogRPCServer) GetLeavesByHash(ctx context.Context, req *trillia
 		return nil, err
 	}
 
-	tree, ctx, err := t.getTreeAndContext(ctx, req.LogId, optsLogRead)
+	tree, hasher, err := t.getTreeAndHasher(ctx, req.LogId, optsLogRead)
 	if err != nil {
 		return nil, err
 	}
+	ctx = trees.NewContext(ctx, tree)
+
+	for i, hash := range req.LeafHash {
+		if got, want := len(hash), hasher.Size(); got != want {
```

Comment:
There's a validateLeafHash() method in that file that currently just checks `len(hash) != 0`. Maybe we could move this check into there and the same checks would apply for both hash related API calls.

Reply:
I did that first, but it required slightly more pervasive changes because that is currently called before the tree is fetched, which is required for the hash size check. I'll add a commit with this refactor and you can decide whether you prefer it.